internal/refactoring/remove_statement.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package refactoring
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/opentofu/opentofu/internal/addrs"
+	"github.com/opentofu/opentofu/internal/configs"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+)
+
+type RemoveStatement struct {
+	From      addrs.ConfigRemovable
+	DeclRange tfdiags.SourceRange
+}
+
+// GetEndpointsToForget recurses through the modules of the given configuration
+// and returns an array of all "removed" addresses within, in a
+// deterministic but undefined order.
+// We also validate that the removed modules/resources configuration blocks were removed.
+func GetEndpointsToForget(rootCfg *configs.Config) ([]addrs.ConfigRemovable, tfdiags.Diagnostics) {
```

Comment:
We seem to flip between Remove and `Forget` here, and I understand there's a nuance, but It's a bit confusing when we're calling it `Removable` here.

Maybe we should just stick to `GetEndpointsToRemove`?